var/spack/repos/builtin/packages/moab/package.py

```diff
@@ -62,6 +68,10 @@ class Moab(AutotoolsPackage):
     # depends_on('cgns', when='+cgns')
     # depends_on('vtk', when='+vtk')
 
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('m4', type='build')
```

Comment:
I would add `when='@master,5.0.1:'` to all of these.

@tgamblin It would be interesting if we could add some kind of `when='git'` or `git=True` tag to `depends_on` so that dependencies could be added when installing from a git repo.